<launch>
  <arg name="hand_gripper_mode" default="false"/>
  <arg name="input_image" default="/d435_head/color/image_rect_color"/>
  <arg name="camera_info" default="/d435_head/color/camera_info"/>
  <arg name="joint_states" default="/robot/joint_states"/>
  <arg name="robot_urdf" default="$(find virtual_robot_ros)/data/urdf/baxter.urdf"/>
  <arg name="base_frame" default="base"/>
  <arg name="hand_dist" default="/eus_imitation/hand_joy_node/hand_distance" />

  <!-- whole robot body render -->
  <node name="robot_render_node"
        pkg="virtual_robot_ros" type="robot_render_node.py"
        output="screen" unless="$(arg hand_gripper_mode)" >
    <remap from="~input_image" to="$(arg input_image)" />
    <remap from="~camera_info" to="$(arg camera_info)" />
    <remap from="~joint_states" to="$(arg joint_states)" />
    <remap from="~hand_dist" to="$(arg hand_dist)" />
    <rosparam subst_value="true" >
        robot_urdf: $(arg robot_urdf)
        base_frame: $(arg base_frame)
    </rosparam>
  </node>

  <!-- hand gripper render mode -->
  <node name="gripper_render_node"
        pkg="virtual_robot_ros" type="gripper_render_node.py"
        output="screen" if="$(arg hand_gripper_mode)" >
    <remap from="~input_image" to="$(arg input_image)" />
    <remap from="~camera_info" to="$(arg camera_info)" />
    <remap from="~hand_dist" to="$(arg hand_dist)" />
    <rosparam subst_value="true" >
        robot_urdf: $(arg robot_urdf)
        base_frame: $(arg base_frame)
    </rosparam>
  </node>
</launch>
